src/config.rs

```diff
@@ -57,45 +58,32 @@ impl Default for Config {
 }
 
 impl Config {
+    pub(crate) fn new(value : Value) -> Self {
+        Config {
+            kind : ConfigKind::default(),
+            cache : value
+        }
+    }
+
     /// Merge in a configuration property source.
+    #[deprecated(since = "0.12.0", note = "please use 'ConfigBuilder' instead")]
     pub fn merge<T>(&mut self, source: T) -> Result<&mut Config>
     where
         T: 'static,
         T: Source + Send + Sync,
     {
-        match self.kind {
-            ConfigKind::Mutable {
-                ref mut sources, ..
-            } => {
-                sources.push(Box::new(source));
-            }
-
-            ConfigKind::Frozen => {
-                return Err(ConfigError::Frozen);
-            }
-        }
-
+        self.push(source)?;
```

Comment:
Why are you changing this implementation? There's no need to in this PR, AFAICS.

Reply:
Reverted to the old state.